test/pre-commit.bats

```diff
@@ -60,3 +60,20 @@ load test_helper
   [ "${lines[1]}" == "failure1.txt:1:another line... forbidden" ]
   [ "${lines[2]}" == "failure2.txt:1:me" ]
 }
+
+@test "Runs safely with args beyond the system argument length limit" {
+  setup_good_repo
+  repo_run git-secrets --install $TEST_REPO
+  cd $TEST_REPO
+
+  FILENAME_LENGTH="$(getconf NAME_MAX .)"
+  (( FILE_COUNT = "$(getconf ARG_MAX)" / "$FILENAME_LENGTH" ))
```

Comment:
Perhaps with `+ 1` to be extra sure it hits the limit?

Reply:
Sure. I was counting on the invocation itself to push it over, but with 255ch filenames that's not so certain. No harm in doing a few more.